doc/source/install/index.rst

```diff
@@ -2,34 +2,330 @@
 Installation
 ============
 
-Each OS version has it's own package names, versions, bugs. However there is a repository,
-which tries to address those issues (at least for RPM based distributions) - https://build.opensuse.org/project/show/Cloud:OTC:Tools:OpenStack.
-RPMs for most often used OSs are prepared and can be taken there. 
+There are several ways to install python-otcextensions to enhance the native ``openstack`` CLI client and to extend the OpenStack SDK to cover the additional Open Telekom Cloud services providing a larger functionality on top of OpenStack.
 
-With a pip package can be installed the following way (please remember, it is not the latest development state). At the command line::
+The easiest way is to use the Python pip installer which is working distribution independent and can be used in an isolated virtual environment.
+There are also ready-made installation packages for various operating systems which have their own versions, package names and sometimes bugs. A repository based on openSUSE's build services tries to cover these issues which is available under: https://build.opensuse.org/project/show/Cloud:OTC:Tools:OpenStack.
 
-    $ sudo yum install python-pip
-    $ sudo pip install otcextensions --upgrade
+Package Overview
+----------------
 
-Or, if you have virtualenv wrapper installed (more suitable for development)::
+**openstacksdk:** 
+  A library on the client side that translates API calls to an OpenStack cloud into Python function calls.
+**otcextensions:** 
+  An addition to openstacksdk with functionality that is specific for the Open Telekom Cloud.
+**openstackclient:** 
+  An application that turns the Python interface of openstacksdk and otcextensions into a CLI tool.
 
-    $ virtualenv venv
+Installation with pip installer
+-------------------------------
+
+All three packages are written in Python and stored as Python libraries in the PyPi repository. The following section describes the installation of ``openstacksdk``, ``otcextensions`` and ``openstackclient``. Please remember, it is not the latest development state. For this purpose the latest sources needs to be installed.
+
+Installation under Ubuntu or Debian
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After a new installation: a C compiler, Python3 with developer libraries, and package manager pip are required:
+
+.. code-block:: bash
+
+   $ sudo apt update
+   $ sudo apt install gcc python3 python3-dev python3-pip libssl-dev
+
+Now, install all libraries and programs at once with the Python package manager pip. The --user flag provides user wide installation instead of a global installation.
+
+.. code-block:: bash
+
+    $ sudo pip3 install otcextensions openstackclient
+    $# pip3 install otcextensions python-openstackclient --user
```

Comment:
What is this commented line doing here? That's either sudo or --user. Let us mention, that sudo will install package globally, but this is not a preferred option, --user install it for single user (preferred)